android local evaluation (#30)
amplitude/amplitude-dev-center: docs/experiment/sdks/android-sdk.md

        }
        ```

    === "Third Party Analytics"

        4. [Initialize the experiment client](#initialize)

>         5. [Start the SDK with the user](#fetch)

Reviewer: should map to `#start`?

Author: ```suggestion
        5. [Start the SDK with the user](#start)
```

The initializer returns a singleton instance, so subsequent initializations for the same instance name will always return the initial instance. To create multiple instances, use the `instanceName` [configuration](#configuration).

=== "Amplitude Analytics"

    ```kotlin
    val experiment = Experiment.initializeWithAmplitudeAnalytics(
        context, "DEPLOYMENT_KEY", ExperimentConfig()
    )
    ```

=== "Third Party Analytics"

    ```kotlin
    val experiment = Experiment.initialize(context, "DEPLOYMENT_KEY", ExperimentConfig())
    ```

#### Configuration

> The SDK client can be configured once on initialization.

Reviewer 2: ```suggestion
SDK client configuration occurs during initialization.
```

    | <div class="big-column">Name</div> | Description | Default Value |
    | --- | --- | --- |
    | `debug` | Enable additional debug logging within the SDK. Should be set to false in production builds. | `false` |
    | `fallbackVariant` | The default variant to fall back if a variant for the provided key doesn't exist. | `{}` |
    | `initialVariants` | An initial set of variants to access. This field is valuable for bootstrapping the client SDK with values rendered by the server using server-side rendering (SSR). | `{}` |
    | `serverZone` | Select the Amplitude data center to get flags and variants from | `ServerZone.US` |
    | `serverUrl` | The host to fetch remote evaluation variants from. For hitting the EU data center, use `serverZone`. | `https://api.lab.amplitude.com` |
    | `flagsServerUrl` | The host to fetch local evaluation flags from. For hitting the EU data center, use `serverZone`. | `https://flag.lab.amplitude.com` |
    | `fetchTimeoutMillis` | The timeout for fetching variants in milliseconds. | `10000` |
    | `retryFetchOnFailure` | Whether to retry variant fetches in the background if the request doesn't succeed. | `true` |

>     | `automaticExposureTracking` | If true, calling [`variant()`](#variant) will track an exposure event through the configured `exposureTrackingProvider`. If no exposure tracking provider is set, this configuration option does nothing.  | `true` |

Reviewer 2: ```suggestion
    | `automaticExposureTracking` | If true, calling [`variant()`](#variant) tracks an exposure event through the configured `exposureTrackingProvider`. If no exposure tracking provider is set, this configuration option does nothing.  | `true` |
```

    | `fetchOnStart` | If true, always [fetch](#fetch) remote evaluation variants on [start](#start). If false, never fetch on start. If null, dynamically determine whether to fetch on start. | `null` |
    | `pollOnStart` | Poll for local evaluation flag configuration updates once per minute on [start](#start). | `true` |

>     | `automaticFetchOnAmplitudeIdentityChange` | Only matters if you use the `initializeWithAmplitudeAnalytics` initialization function to seamlessly integrate with the Amplitude Analytics SDK. If `true` any change to the user ID, device ID or user properties from analytics will trigger the experiment SDK to fetch variants and update it's cache. | `false` |

Reviewer 2: ```suggestion
    | `automaticFetchOnAmplitudeIdentityChange` | Only matters if you use the `initializeWithAmplitudeAnalytics` initialization function to seamlessly integrate with the Amplitude Analytics SDK. If `true` any change to the user ID, device ID or user properties from analytics triggers the experiment SDK to fetch variants and update it's cache. | `false` |
```

    | `userProvider` | An interface used to provide the user object to `fetch()` when called. See [Experiment User](https://developers.experiment.amplitude.com/docs/experiment-user#user-providers) for more information. | `null` |
    | `exposureTrackingProvider` | Implement and configure this interface to track exposure events through the experiment SDK, either automatically or explicitly. | `null` |
    | `instanceName` | Custom instance name for experiment SDK instance. **The value of this field is case-sensitive.** | `null` |

> 

Reviewer: Missing `source` config

!!!info "EU Data Center"

>     If you're using Amplitude's EU data center, configure the `serverZone` option on initialization to `ServerZone.EU`.

Reviewer 2: ```suggestion
    If you use Amplitude's EU data center, configure the `serverZone` option on initialization to `ServerZone.EU`.
```

The initializer returns a singleton instance, so subsequent initializations for the same instance name will always return the initial instance. To create multiple instances, use the `instanceName` [configuration](#configuration).

=== "Amplitude Analytics"

    ```kotlin
    val experiment = Experiment.initializeWithAmplitudeAnalytics(
        context, "DEPLOYMENT_KEY", ExperimentConfig()
    )
    ```

=== "Third Party Analytics"

    ```kotlin
    val experiment = Experiment.initialize(context, "DEPLOYMENT_KEY", ExperimentConfig())
    ```

#### Configuration

SDK client configuration occurs during initialization.

???config "Configuration Options"
    | <div class="big-column">Name</div> | Description | Default Value |
    | --- | --- | --- |
    | `debug` | Enable additional debug logging within the SDK. Should be set to false in production builds. | `false` |
    | `fallbackVariant` | The default variant to fall back if a variant for the provided key doesn't exist. | `{}` |
    | `initialVariants` | An initial set of variants to access. This field is valuable for bootstrapping the client SDK with values rendered by the server using server-side rendering (SSR). | `{}` |

>     | `source` | The primary source of variants. Set the value to `Source.InitialVariants` and configure `initialVariants` to bootstrap the SDK for SSR or testing purposes. | `Source.LocalStorage` |

Reviewer: `Source.LOCAL_STORAGE` and `Source.INITIAL_VARIANTS`

Author: local storage is the default.

        ExperimentConfig config = ExperimentConfig.builder()
            .exposureTrackingProvider(new SegmentExposureTrackingProvider(analytics))
            .build();

>         ExperimentClient experiment = Experiment.initialize(context, "<DEPLOYMENT_KEY>", config);

Reviewer: do we need to remove angle brackets for `<FLAG_KEY>` as well?

                .userId(userId)
                .deviceId(deviceId)
                .build()
            experiment.start(user).get()
        } catch (e: Exception) {
            e.printStackTrace()
        }
        ```

---

### Start

> Start the SDK by getting flag configurations from the server and fetching remote evaluation variants for the user. The SDK is ready once the returned future resolves.

Reviewer 2: ```suggestion
The Experiment SDK starts when you get flag configurations from the server and fetch remote evaluation variants for the user. The SDK is ready after the returned future resolves.
```

Author: ```suggestion
Start the Experiment SDK to get flag configurations from the server and fetch remote evaluation variants for the user. The SDK is ready once the returned future resolves.
```
